File: Assignment22_2.java
import java.util.Scanner;
/////////////////////////////////////////////////////////////////////////////////
//    Function Name :       FrequencyDifference
//    Description :         To get count of even number from the array
//    Input :               Integer
//    Output :              Integer
//    Author :              Raj Samir Jadhav
//    Date :                16/11/2025
////////////////////////////////////////////////////////////////////
public class FrequencyDifference
{
    public static int Frequency(int arr[], int length)              // Function to return difference between frequency of even and odd numbers
    {
        int evenCount = 0, oddCount = 0;

        for(int i = 0; i < length; i++)                              // Loop through all elements
        {
            if(arr[i] % 2 == 0)                                      // Check even
            {
                evenCount++;
            }
            else                                                      // Odd number
            {
                oddCount++;
            }
        }

        return evenCount - oddCount;                                  // Return difference (Even - Odd)
    }

    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of elements: ");                       // Ask user for number of elements
        int size = sc.nextInt();

        int arr[] = new int[size];

        System.out.println("Enter " + size + " elements:");                    // Input array elements
        for(int i = 0; i < size; i++)
        {
            System.out.print("Enter element " + (i + 1) + ": ");
            arr[i] = sc.nextInt();
        }

        int result = Frequency(arr, size);                                    // Call function

        System.out.println("Difference (Even - Odd) = " + result);

        sc.close();
    }
}

File: Assignment22_2.c
#include <stdio.h>
#include <stdlib.h>

/////////////////////////////////////////////////////////////////////////////////
//    Function Name :       Frequency
//    Description :         To caculate frequency btw even and odd integer
//    Input :               Integer
//    Output :              Integer
//    Author :              Raj Samir Jadhav
//    Date :                16/11/2025
////////////////////////////////////////////////////////////////////

int Frequency(int Arr[], int iLength)                         // Function to calculate difference between frequency of even and odd numbers
{
    int iCnt = 0;
    int iEven = 0, iOdd = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        if(Arr[iCnt] % 2 == 0)                                // If number is even
        {
            iEven++;
        }
        else                                                 // If number is odd
        {
            iOdd++;
        }
    }

    
    return iEven - iOdd;
}

int main()
{
    int iSize = 0, iRet = 0, iCnt = 0;
    int *p = NULL;

    
    printf("Enter number of elements: ");                   // Ask user for number of elements
    scanf("%d", &iSize);

    p = (int *)malloc(iSize * sizeof(int));                // Allocate memory

    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    
    printf("Enter %d elements:\n", iSize);
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element %d: ", iCnt + 1);
        scanf("%d", &p[iCnt]);
    }

    iRet = Frequency(p, iSize);                                 // Function call


    printf("Difference (Even - Odd) = %d\n", iRet);

    
    free(p);                                                    // Free allocated memory

    return 0;
}
